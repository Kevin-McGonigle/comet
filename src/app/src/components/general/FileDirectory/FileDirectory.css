.fileDirectory {
    background: #11182f;
    color: #fff;
    height: 100%;
    padding: 20px;
    width: 250px;
    left: 0;
    top: 0;
    position: absolute;
    border-right: 1px solid white;
    overflow: hidden;
}

.fileDirTitle {
    margin-bottom: 10px;
}

.fileDirLogo {
    display: flex;
    flex-direction: row;
    align-items: center;
}

.fileDirTable {
    width: 250px;
    display: flex;
    flex-direction: column;
}

