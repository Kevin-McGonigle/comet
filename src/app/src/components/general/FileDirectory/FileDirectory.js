import React, { useState } from 'react';
import styles from './FileDirectory.css';
import comet from '../../Homepage/comet.png';
import classnames from 'classnames';
import { Table, Pane, Checkbox } from 'evergreen-ui';

const cx = args => classnames(styles, args)

const FileDirectory = props => {
    const {
        fileNames,
        setSelectedFile,
    } = props;

    const setSelectedFileHandler = (event) => {
        const fileName = event.currentTarget.textContent;
        setSelectedFile(fileName);
    }
    
    return (
        <div className={cx('fileDirectory')}>
            <div className={cx('fileDirLogo')}>
                Comet
                <img src={comet} width="50px" height="50px" />
            </div>
            
            <div className={cx('title')}>File Directory</div>

            <div className={cx('fileDirTable')}>
                <Table>
                    <Table.Head>
                        <Table.SearchHeaderCell />
                    </Table.Head>

                    <div className={cx('fileItemContainer')}>
                        <Pane>
                            {fileNames && fileNames.map(file => {
                                return (
                                    <Table.Row
                                        key={file}
                                        isSelectable
                                        isSelected
                                        width="250px"
                                        textAlign="center"
                                        isHighlighted={true}
                                        onClick={e => setSelectedFileHandler(e)}
                                    >
                                        <Table.TextCell id={file}>{file}</Table.TextCell>
                                    </Table.Row>
                                )
                            })}
                        </Pane>
                    </div>
                </Table>
            </div>
        </div>
    )
};

export default FileDirectory;