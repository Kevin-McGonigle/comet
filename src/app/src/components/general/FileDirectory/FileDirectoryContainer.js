import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import FileDirectory from './FileDirectory';
import { actions as fileActions } from '../../../store/fileData/fileData';

const mapStateToProps = state => ({
    files: state.files.files,
});

const mapDispatchToProps = dispatch => bindActionCreators(fileActions, dispatch);

export const FileDirectoryContainer = connect(mapStateToProps, mapDispatchToProps)(FileDirectory);



