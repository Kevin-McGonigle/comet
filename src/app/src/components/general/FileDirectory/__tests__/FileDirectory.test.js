import React from 'react';
import FileDirectory from '../FileDirectory';
import { shallow, mount } from 'enzyme';
import "../../../../setupTests"

const fileNames = ["FileName1.py", "FileName2.py"];

test('should render FileDirectory succesfully', () => { 
    const component = shallow(<FileDirectory fileNames={fileNames}/>);
    expect(component.html()).toMatchSnapshot();
});

test('should call setSelectedFile upon itemFile click', () => { 
    const setSelectedFile = jest.fn();
    const component = shallow(<FileDirectory fileNames={fileNames} setSelectedFile={setSelectedFile}/>);
    const button = component.find({ id: 'FileName2.py' }).parent();
    button.simulate('click', { currentTarget: { textContent: "FileName2.py" }});
    expect(setSelectedFile).toHaveBeenCalled();
});

