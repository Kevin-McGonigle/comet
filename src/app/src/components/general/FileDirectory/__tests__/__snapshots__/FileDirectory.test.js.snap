// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FileDirectory succesfully 1`] = `"<div class=\\"fileDirectory\\"><div class=\\"fileDirLogo\\">Comet<img src=\\"comet.png\\" width=\\"50px\\" height=\\"50px\\"/></div><div class=\\"fileDirTitle\\">File Directory</div><div class=\\"fileDirTable\\"><div class=\\"css-nil ðŸ“¦box-szg_border-box\\"><div class=\\"css-nil ðŸ“¦b-btm_1px-solid-E4E7EB ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-srnk_0 ðŸ“¦pr_0px ðŸ“¦h_32px ðŸ“¦box-szg_border-box\\"><div class=\\"css-saeo60 css-nil ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_1 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_visible ðŸ“¦ovflw-y_visible\\"><svg style=\\"fill:#7B8B9A\\" data-icon=\\"search\\" viewBox=\\"0 0 16 16\\" class=\\"ðŸ“¦ml_2px ðŸ“¦mr_10px ðŸ“¦w_12px ðŸ“¦h_12px ðŸ“¦box-szg_border-box\\"><path d=\\"M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z\\" fill-rule=\\"evenodd\\"></path></svg><input class=\\"css-q4k7k7 ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ml_-2px ðŸ“¦pl_0px ðŸ“¦box-szg_border-box\\" spellcheck=\\"true\\" placeholder=\\"Filter...\\"/></div> <div aria-hidden=\\"true\\" style=\\"position:fixed;top:-500px;left:-500px;width:100px;overflow-y:scroll\\"><div></div></div></div><div class=\\"fileItemContainer\\"><div class=\\"css-nil ðŸ“¦box-szg_border-box\\"><div class=\\"css-z3ro6d css-nil ðŸ“¦b-btm_1px-solid-EDF0F2 ðŸ“¦dspl_flex ðŸ“¦h_48px ðŸ“¦w_250px ðŸ“¦txt-algn_center ðŸ“¦box-szg_border-box\\" aria-selected=\\"false\\" aria-current=\\"true\\" data-isselectable=\\"true\\" tabindex=\\"-1\\"><div class=\\"css-saeo60 css-nil ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_1 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦bg-clr_grey\\" id=\\"FileName1.py\\"><span class=\\"ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦txt-ovrf_ellipsis ðŸ“¦wht-spc_nowrap ðŸ“¦box-szg_border-box\\">FileName1.py</span></div></div><div class=\\"css-z3ro6d css-nil ðŸ“¦b-btm_1px-solid-EDF0F2 ðŸ“¦dspl_flex ðŸ“¦h_48px ðŸ“¦w_250px ðŸ“¦txt-algn_center ðŸ“¦box-szg_border-box\\" aria-selected=\\"false\\" aria-current=\\"true\\" data-isselectable=\\"true\\" tabindex=\\"-1\\"><div class=\\"css-saeo60 css-nil ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box ðŸ“¦flx_1 ðŸ“¦dspl_flex ðŸ“¦algn-itms_center ðŸ“¦flx-srnk_0 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦bg-clr_grey\\" id=\\"FileName2.py\\"><span class=\\"ðŸ“¦color_425A70 ðŸ“¦fnt-fam_b77syt ðŸ“¦fnt-sze_12px ðŸ“¦f-wght_400 ðŸ“¦ln-ht_16px ðŸ“¦ltr-spc_0 ðŸ“¦flx_1 ðŸ“¦ovflw-x_hidden ðŸ“¦ovflw-y_hidden ðŸ“¦txt-ovrf_ellipsis ðŸ“¦wht-spc_nowrap ðŸ“¦box-szg_border-box\\">FileName2.py</span></div></div></div></div></div></div></div>"`;
